.github/workflows/react-build.yml

```diff
@@ -0,0 +1,39 @@
+# This workflow will do a clean installation of node dependencies, cache/restore them, build the source code and run tests across different versions of node
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-nodejs
+
+name: React.js Build CI
+
+on:
+  push:
+    branches-ignore:
+      - main
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches:
+      - main
+
+jobs:
+  build:
+    defaults:
+      run:
+        working-directory: gui
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [18.x]
+        # See supported Node.js release schedule at https://nodejs.org/en/about/releases/
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v3
+      with:
+        node-version: ${{ matrix.node-version }}
+        cache: 'npm'
+        cache-dependency-path: gui/package-lock.json
+    - run: npm install
```

Comment:
Should we be running `npm ci` rather than `npm install` to leverage the package-lock.json? I think that will result in a more consistent build process.

Reply:
Done.